### ISSUE TITLE
AAL roll-forward never reconciles to the direct calculation after year 1 (gain/loss stubbed to 0) — surfaced by ad-hoc COLA scenario

### ISSUE THREAD

**Issue by donboyd5** [donboyd5/pension_model#410]
## Summary

The model carries the AAL two ways, and reconciles them only in the first projection year:

1. **Direct calculation** (`*_est` liability columns): each year, recompute the present value of all future benefits from the projected census. A fresh valuation every year.
2. **Roll-forward** (`aal_{tier}`, the value reported in `summary.aal` and used for unfunded liability, funded ratio, and — for actuarially-funded plans — contributions): start from last year's balance, add interest and normal cost, subtract benefits, plus a `liability_gain_loss` adjustment.

The roll-forward is seeded from the direct calculation in year 1 (`_funding_setup.py:492`), then runs on its own. `liability_gain_loss` — the term that would re-true the roll-forward back to the fresh valuation — is **hard-set to 0** (`pipeline.py:618` per tier, `:624` total). The only built-in check (`identity_checks.py`) verifies the roll-forward is self-consistent, **not** that it matches the direct calculation. So after year 1 nothing pulls the two back together, and they drift.

In a deterministic projection the assumptions *are* the experience, so the gain/loss should be exactly 0 **only if** the roll-forward reproduces the direct calculation each year. It does not — they diverge by 17–29% by year 30 across plans (FRS roll-forward ends +26% above its direct value; alers −29%; arpers −17%).

## How it surfaced — the ad-hoc COLA scenario (#369, PR #405)

PR #405 (`retiree_ad_hoc_cola_2pct`) books the catch-up `liability_gain_loss` **only in the enactment year**. From then on the roll-forward keeps subtracting the extra COLA benefit payments with no matching liability credit, so the reported COLA AAL increment sinks and eventually flips negative — impossible for a benefit increase.

Reproduced on `arpers` (baseline vs scenario), COLA AAL increment, directly-computed vs funding-reported:

| Year | Directly-computed | Funding-reported |
|------|------------------:|-----------------:|
| 2030 | +$128.4M | +$126.3M |
| 2035 | +$102.3M | +$96.6M |
| 2045 | +$49.4M | +$32.9M |
| 2055 | **+$14.7M** | **−$20.6M** (negative from 2051 on) |

FRS shows the same pattern (+$2.43B at enactment → funding-reported −$1.49B by 2052 vs correct +$0.31B), and for FRS the under-stated AAL also contaminates contributions. Benefit payments, the untouched valuation year, and the directly-computed liability are all correct. The PR's tests pass because test #3 checks only the directly-computed table and test #4 checks only the enactment year; nothing tests the multi-year roll-forward.

## Fix — two parts, very different sizes

**Part 1 — narrow, urgent, scenario-only (no R-baseline impact).** Book the COLA increment's gain/loss in every year from enactment on, not just the first:

```
gl_t = increment_t − increment_{t-1}·(1+dr) + extra_benefit_t·(1+dr)^0.5
```

This collapses to PR #405's current formula in the enactment year and adds the missing piece afterward, making the scenario's marginal impact correct. It touches only the COLA increment, so it does not move baseline or R-matched results.

**Part 2 — broad, deliberate, phase-post-r (moves R baselines).** The base liability has the same unreconciled drift. Two clean options:
- Use the direct calculation as the liability source each year (drop the roll-forward as the reported AAL); the drift disappears, OR
- Keep the roll-forward but set `liability_gain_loss = direct − rolled` every year so it reconciles (Part 1 generalized to the whole liability).

Either prevents this class of bug. This changes FRS/TXTRS funded-ratio paths, so it needs deliberate re-baselining — not a silent edit.

## Pointers
- Seed: `_funding_setup.py:492` · gain/loss stub: `pipeline.py:618,624` · roll-forward: `_funding_phases.py:404-423` (`_phase_liability_gl_and_aal`) · self-consistency check only: `identity_checks.py`
- Implements/related: #369 (COLA enacted after valuation date), #277 (unify funding-side payroll with open-group projection), #283 (AVA gain/loss recognition)
- Phase: `phase-post-r` (the root cause and Part 2 change R behavior; Part 1 is a safe sub-task that can land immediately now that the post-R queue is active).

**Comment by donboyd5** [comment]
## Status check: Part 1 landed, and Part 2 is now smaller than this issue says

**Part 1 is done.** Commit `470e64d`, 2026-06-23 — *"fix(funding): book one-time COLA gain/loss every year, not just at enactment"*. `_funding_phases._phase_liability_gl_and_aal` carries exactly the prescribed term:

```python
gl = gl + incr - incr_prev * (1 + dr) + extra_ben * (1 + dr) ** 0.5
```

`test_one_time_cola_scenario.py` covers it, 6 passed. The issue never recorded it.

## Part 2: FRS is already fixed, as a side effect of the DROP migration

The issue's second clean option was *"keep the roll-forward but set `liability_gain_loss = direct − rolled` every year so it reconciles."* #436 built that for the status-engine path, with one deliberate difference: the residual goes into its own `aal_reconciliation_{tier}` column rather than into `liability_gain_loss`, so a term that is otherwise exactly zero does not quietly absorb it.

Since FRS migrated to the engine (#448), its reported AAL **is** the direct valuation every year.

## Re-measured on current `main`

Reported AAL against the sum of the classes' `total_aal_est`:

| Plan | Year 0 | Year 10 | Year 20 | Final |
|---|---|---|---|---|
| **frs** | 0.0000% | **+0.00%** | **+0.00%** | **+0.00%** |
| txtrs | 0.0000% | +1.57% | +4.18% | **+8.28%** |
| arpers | 0.0000% | −2.06% | −7.74% | **−17.10%** |
| alers | **+5.54%** | −4.36% | −13.20% | **−28.88%** |

arpers (−17%) and alers (−29%) match the figures in the issue body. FRS's +26% is gone. txtrs was not previously cited and is +8.28%.

So the remaining exposure is **txtrs, arpers and alers**, and the fix pattern is no longer hypothetical — it is running in production on FRS.

## One new finding: alers is off at year 0

Every other plan reconciles exactly at the valuation date. **alers's reported AAL starts 5.54% above its own directly-computed value**, before any drift.

That is a different thing from the drift this issue is about — a seeding discrepancy, not a roll-forward one — and it is the largest single number on the table. It should be run down before alers's drift is interpreted, since a wrong starting point makes the year-30 −28.88% partly a level error rather than accumulation. I have not traced it and am not assuming a cause.

## Suggested shape for the rest

Adopt the engine's approach for the other three plans: report the direct valuation and record the difference in `aal_reconciliation_{tier}`, which already exists, already accumulates to the plan frame, and is already named in `identity_checks._check_aal_tier`. That makes the residual visible and boundable rather than hidden in a column that reads zero.

It moves numbers on txtrs, arpers and alers, so it needs the usual which-numbers-move statement and refreshed snapshots — and for txtrs, refreshed R-baseline handling of the kind #444 set up for FRS.

**Comment by donboyd5** [comment]
## Traced: alers's year-0 gap is a calibration gap, not roll-forward drift

The 5.54% I flagged is not drift and not a roll-forward defect. **Every alers class reconciles to its own directly-computed value exactly** — +0.0000% for all three. The gap is entirely at the plan level, and it is a difference in what year 0 reports.

| Frame | Year-0 total AAL | Source |
|---|---|---|
| Sum of the three classes | $9,734,734,041 | the model's own valuation |
| Plan frame, as reported | $10,273,992,786 | `init_funding.csv`, i.e. the AV |
| | **+5.54%** | |

So alers's plan-level year-0 AAL is the **published AV figure**, while its class frames carry the **model's own computation**. `test_consistency` already skips year 0 on the sum-of-classes identity for exactly this reason — *"Year 0 is seed data from the ACFR, not computed from class sums."*

## Why FRS reads 0.00% and alers does not

Not a difference between the plans' data — a difference in path. FRS now runs the status engine, and `_phase_liability_gl_and_aal` reports the engine's direct valuation rather than the roll-forward (#436), so FRS's plan frame carries the model's number at year 0. alers is on the ordinary path and carries the seed.

For the record, FRS's own seed is $217,434,441,000 against a model sum of $218,199,836,288 — **−0.35%**, so FRS would show a small gap on the same basis.

## What the number actually says

**alers's model computes 5.25% less liability than the AV it is seeded from.** Compared at the AV's own reporting granularity — the AV publishes State Employees, State Police and Local Employees, not the runtime's three classes:

| AV reporting group | Model | AV seed | Difference | |
|---|---|---|---|---|
| State Employees | $8,987,251,880 | $9,465,504,482 | −$478,252,602 | **−5.05%** |
| State Police | $747,482,161 | $808,488,304 | −$61,006,143 | **−7.55%** |
| **Total modelled** | **$9,734,734,041** | **$10,273,992,786** | **−$539,258,745** | **−5.25%** |

A per-class split looks worse and is misleading: `state_employees_flc` reads **+18.99%**. But the AV does not publish FLC separately — `prep/alers/input_checklist.csv` records that the runtime splits State Employees into FLC and non-FLC *"using the tier × FLC payroll shares"*, so `init_funding.csv`'s per-class figures for those two are themselves a construction. Comparing the model's FLC liability against a payroll-share split is comparing two constructions, not validating against the AV.

And the direction is what one would expect: FLC members have richer benefits and earlier retirement, so their share of *liability* should exceed their share of *payroll*. The +19% may be substantially right and the split wrong.

## What this means for this issue

- alers's year-30 **−28.88%** is measured against a plan-level series that starts on a different basis from its classes. Part of it is this level difference, not accumulation. The drift should be re-measured class by class, where the bases agree.
- The 5.25% itself belongs to alers, not here. Filed on #350.
